src/datasources/perf-ds/PerformanceHelpers.ts

```diff
@@ -9,93 +9,101 @@ import { OnmsMeasurementsQueryResponse } from "./types";
  * @param end an ending timestamp for the window, or 0 to not limit the end of the window range
  */
 const getWindowedTimestamps = (timestamps: number[], start: number, end: number) => {
-  if (start === 0 && end === 0) {
+    if (start === 0 && end === 0) {
+        return {
+            windowedTimestamps: [...timestamps],
+            startIndex: 0,
+            endIndex: timestamps.length - 1
+        }
+    }
+
+    const windowedTimestamps = [] as number[]
+    let startIndex = -1
+    let endIndex = timestamps.length - 1
+
+    for (let i = 0; i < timestamps.length; i++) {
+        const tsVal = timestamps[i]
+
+        if (end > 0 && tsVal > end) {
+            endIndex = i - 1
+            break
+        }
+
+        if (start === 0 || tsVal >= start) {
+            if (startIndex < 0) {
+                startIndex = i
+            }
+            windowedTimestamps.push(tsVal)
+        }
+    }
+
     return {
-      windowedTimestamps: [...timestamps],
-      startIndex: 0,
-      endIndex: timestamps.length - 1
+        windowedTimestamps,
+        startIndex,
+        endIndex
     }
-  }
-
-  const windowedTimestamps = [] as number[]
-  let startIndex = -1
-  let endIndex = timestamps.length - 1
-
-  for (let i = 0; i < timestamps.length; i++) {
-      const tsVal = timestamps[i]
-
-      if (end > 0 && tsVal > end) {
-          endIndex = i - 1
-          break
-      }
-
-      if (start === 0 || tsVal >= start) {
-          if (startIndex < 0) {
-              startIndex = i
-          }
-          windowedTimestamps.push(tsVal)
-      }
-  }
-
-  return {
-    windowedTimestamps,
-    startIndex,
-    endIndex
-  }
 }
 
 /**
  * Convert QueryResponse data returned by OpenNMS Measurements Rest API to Grafana DataFrame format.
  */
 export const measurementResponseToDataFrame =
-    (measurementData: OnmsMeasurementsQueryResponse, queryRefId?: string): DataFrame => {
+    (measurementData: OnmsMeasurementsQueryResponse): DataFrame[] => {
 
-    const { start, end, labels, columns, timestamps, metadata } = measurementData
+        const { start, end, labels, columns, timestamps, metadata } = measurementData
 
-    const dataFrame: DataFrame = {
-        name: '',
-        refId: queryRefId,
-        length: 0,
-        fields: []
-    }
+        const dataFrames: DataFrame[] = []
 
-    if (!columns || !columns.length || !timestamps || !timestamps.length) {
-        return dataFrame
-    }
+        let dataFrame: DataFrame = getEmptyDataFrame()
 
-    const { windowedTimestamps, startIndex, endIndex } = getWindowedTimestamps(timestamps, start, end)
+        if (!columns || !columns.length || !timestamps || !timestamps.length) {
+            dataFrames.push(dataFrame)
```

Comment:
Here and also a few lines later where we check for empty `windowedTimestamps`, we should probably either just return `dataFrames` with a single empty `dataFrame` and short circuit any other processing

*or*

do not add the empty `dataFrame` here (remove the check and the code), but just fall through to the for loop, which will add a `dataFrame` per `label`, with an empty set of values